osu.Game/Screens/OnlinePlay/Multiplayer/Match/Playlist/MultiplayerPlaylist.cs

```diff
@@ -154,22 +144,35 @@ private void playlistItemChanged(MultiplayerPlaylistItem item) => Schedule(() =>
 
         private void addItemToLists(MultiplayerPlaylistItem item)
         {
-            var apiItem = client.Room?.Playlist.SingleOrDefault(i => i.ID == item.ID);
-
-            // Item could have been removed from the playlist while the local player was in gameplay.
-            if (apiItem == null)
```

Comment:
While this is suspicious, it was added in https://github.com/ppy/osu/commit/740a6f16c79a606652649fc9e71cb607bfb343d9 where we would have to turn `MultiplayerPlaylistItems`s into queries on `Room.Playlist` for `PlaylistItem` conversion. Since we're no longer doing that and instead `new PlaylistItem(item)`, this isn't required.

There's _potentially_ an argument to be made for an optimisation opportunity to requery and not add items if they don't exist (because all calls here are scheduled), but that's minor to me at this stage.